entity-service-client/src/main/java/org/hypertrace/entity/data/service/client/EdsCacheClient.java

```diff
@@ -61,7 +89,16 @@ public Entity upsert(Entity entity) {
   @Override
   public Entity getByTypeAndIdentifyingAttributes(String tenantId,
       ByTypeAndIdentifyingAttributes byIdentifyingAttributes) {
-    return client.getByTypeAndIdentifyingAttributes(tenantId, byIdentifyingAttributes);
+    EdsTypeAndIdAttributesCacheKey key = new EdsTypeAndIdAttributesCacheKey(tenantId,
+        byIdentifyingAttributes);
+    try {
+      String entityId = entityIdsCache.get(key).orElse(null);
```

Comment:
You shouldn't need `orElse` because the cache returns `Optional`

Reply:
Done